core/src/main/scala/libretto/scaletto/ScalettoLib.scala

```diff
@@ -281,27 +278,26 @@ class ScalettoLib[
   : (A |*| B) -⚬ ((A |*| B) |+| (B |*| A)) =
     lteqBy(aKey, bKey).>.right(swap)
 
-  implicit def comparableVal[A](using A: Ordering[A]): Comparable[Val[A], Val[A]] =
-    new Comparable[Val[A], Val[A]] {
-      import coreLib.Compared._
+  given [A : Ordering]: Comparable[Val[A], Val[A]] with {
```

Comment:
I usually prefer context bounds. But didn't see them much around here.

Is this by design?

Reply:
I tend to avoid them because on the call site they are behave like implicits, i.e. if passed explicitly, they are passed using a normal argument list. This leads to problems like this:

```scala
trait Foo[A]

trait Bar[A] {
  def apply(s: String): Unit = println(s)
}

def makeBar1[A](using Foo[A]): Bar[A] = new Bar[A] {}
def makeBar2[A: Foo]         : Bar[A] = new Bar[A] {}

given Foo[Int] with {}

makeBar1[Int]("Hi") // OK
makeBar2[Int]("Hi") // Found:    ("Hi" : String)
                    // Required: Foo[Int]
```
https://scastie.scala-lang.org/Psycho2QRL6dZ9Qa28PLwQ

That said, it's probably harmless in this particular case.